### DISCUSSION TITLE
vc-drawings的样式问题.

### DISCUSSION THREAD

**Discussion by 393209049** [zouyaoji/vue-cesium#473] [Q&A]
通过  import 'vue-cesium/dist/index.css'; 这种方式引入会造成全局污染,其他UI组件的样式错乱.
通过 <style lang="css" src="vue-cesium/dist/index.css" scoped></style> 这种方式引入没有效果.
另外我单独创建了一个vue项目单独引入vue-cesium,vc-drawings也是在地图外边.
![1700287734812](https://github.com/zouyaoji/vue-cesium/assets/34265928/2afd6552-d93d-4015-bf04-85aeb5e001d5)
通过,width:100vw,height:100vh让地图全屏,vc-drawings倒是在地图里.

**Comment by zouyaoji** [comment]
你用的啥UI库？要找出哪个样式影响了你的样式才行。按理不应该啊，大部分都是加了vc-开头的

**Reply by 393209049** [reply]
![image](https://github.com/zouyaoji/vue-cesium/assets/34265928/1a603d94-5215-4228-a074-d5e39e4d72bb)
比如这个.flex样式,在我项目的其他UI组件也用到了这个样式名.结果这个里边有flex-wrap: wrap,导致其他UI换行错位.

**Reply by 393209049** [reply]
之前那个vc-drawings的UI按钮在地图外边的问题,最终只需要在vc-viewer的标签上加上style="position: relative",就好了.

**Reply by zouyaoji** [reply]
意思是搞定了嘛？

如果覆盖了其他UI库的样式，有个简单的办法就是，先导入vue-cesium的样式，后导入其他UI库的样式，这样就以其他UI库的样式为准了哈。

**Reply by 393209049** [reply]
暂时就发现这个.flex{display:flex;flex-wrap:wrap}造成的样式错乱,我在APP.vue中用:deep(.flex){flex-wrap:noWrap}覆盖了.因为css在同class名,但参数不同时会合并,因此先导入vue-cesium的样式,再导入其他的,对于类似这个.flex的问题不管用.

**Reply by wssnail** [reply]
> ![image](https://private-user-images.githubusercontent.com/34265928/284120262-1a603d94-5215-4228-a074-d5e39e4d72bb.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzIwODE4NTEsIm5iZiI6MTczMjA4MTU1MSwicGF0aCI6Ii8zNDI2NTkyOC8yODQxMjAyNjItMWE2MDNkOTQtNTIxNS00MjI4LWEwNzQtZDVlMzllNGQ3MmJiLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDExMjAlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMTIwVDA1NDU1MVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWU4MDE2Y2VkODA2YWIwZWFmMzM5YmIyYWNhNGJkY2Q1NTdjMmQ3Y2IyOWFhN2IwOTU0NjdmMDczYWM4Nzg1MmEmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.gPDyhNaSKDDP-jbM1P7bXZ-q5Qn_5QYUi0H8QMf9AvU) 比如这个.flex样式,在我项目的其他UI组件也用到了这个样式名.结果这个里边有flex-wrap: wrap,导致其他UI换行错位.

我也遇到了同样的问题，在main.ts里引入css导致.flex级别太高，影响了windicss的flex

**Reply by zouyaoji** [reply]
@wssnail 

目前建议先引入vue-cesium的css，后引入你其他的css。

**Reply by wssnail** [reply]
> @wssnail
> 
> 目前建议先引入vue-cesium的css，后引入你其他的css。

试过了，对windicss不行，在单独的vue文件中使用都会出现问题
